### PR TITLE
Feature/menu component showcase

### PR DESCRIPTION
**Pull request by AliNisarAhmed**
## Description

(NOTE: the PR is currently WIP)

Adds Menu component usage examples to showcase

## Action Items

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] This PR is as small as it can be, and only implements on thing (not many)
- [ ] I have manually tested my feature
- [ ] I have written a "good" amount of tests (including visual and unit tests)

### CODE REVIEW

**Comment by AliNisarAhmed** [comment]
@supermacro The PR is WIP. I created it so that I can discuss and get guidance.

Is the Menu API fully implemented? I did not see a way to make the button disabled, nor did I see a way to change the MenuMode, so I added those two functions to Ant.Menu. Am I missing something? Even when I disabled the Menu item it was not actually disabled.

**Comment by supermacro** [comment]
Hey @AliNisarAhmed, 

Sorry for the late response. I haven't yet taken a look, but I will make sure to do so sometime this week.

> is the Menu API fully implemented?

No, but what currently exists is fully usable. The menu on the left-hand-side of the [Component Showcase](https://elm-antd.netlify.app/) is built using `Ant.Menu`.

Thanks for adding additional functionality!

> Am I missing something? Even when I disabled the Menu item it was not actually disabled.

Hmm, off the top of my head I am not sure what's going on. But I'll have a look this week.

**Comment by AliNisarAhmed** [comment]
@supermacro I encountered a few problems just implementing the Horizontal Menu example, so I have taken some liberties to pass state items as params to some functions to display states like `disabled` and `opened` for subMenu, for example. Please review the changes carefully and let me know if they are acceptable, or indicate where change is required.

Right now, if subMenu is "enabled", the list item extend downward and push the other items down with them (instead of hovering above other items). Did you have any plans to implement the hover effects on subMenu as done by Ant D [here](https://ant.design/components/menu/#components-menu-demo-horizontal)?

File: src/Ant/Menu.elm [hunk 1]
```diff
@@ -66,6 +69,17 @@ selected (MenuItem msg currentState contents) =
...
 
+{-| disable the menu item
+-}
+disabled : MenuItem msg -> MenuItem msg
+disabled (MenuItem msg currentState contents) =
+    let
+        newState =
+            { currentState | disabled = True }
+    in
+    MenuItem msg newState contents
```
Comment (supermacro) [inline src/Ant/Menu.elm:80]: 👍

File: src/Ant/Menu.elm [hunk 2]
```diff
@@ -112,6 +126,17 @@ initMenu =
...
 
+{-| Change mode of the Menu
+-}
+mode : MenuMode -> Menu msg -> Menu msg
+mode newMode (Menu config contents) =
+    let
+        newConfig =
+            { config | mode = newMode }
+    in
+    Menu newConfig contents
```
Comment (supermacro) [inline src/Ant/Menu.elm:137]: 👍

**Comment by supermacro** [comment]
Note that there is currently a border on the right hand side of the menu which should not exist:

Screen Shot 2020-08-24 at 5 03 45 PM

Secondly, note that the styles on the active menu item for horizontal menus are different than the ones currently implemented in React's Ant.Design.

It seems like active menus have different styles for horizontal vs vertical menus.

**Comment by supermacro** [comment]
Also, I tried hovering over the submenu item but the submenu did not appear. 

> Did you have any plans to implement the hover effects on subMenu as done by Ant D here?

Hmmm, good question. Maybe in the future, but not a high priority right now.

**Comment by supermacro** [comment]
Hey @AliNisarAhmed, this PR is a bit stale now (the `Container` api has changed quite a bit). If you want to continue working on this, please rebase on top of master.

I will close this PR for now. But feel free to re-open in the future!

**Comment by AliNisarAhmed** [comment]
Sorry I have been busy in some exam, could not finish the feature. I will pick up this feature again during Hacktoberfest and create a fresh PR.

**Pull request closed without merging**